build-tools-internal/src/main/groovy/elasticsearch.bc-upgrade-test.gradle

```diff
@@ -15,4 +15,11 @@ tasks.register("bcUpgradeTest", StandaloneRestIntegTestTask) {
   usesBwcDistribution(Version.fromString("0.0.0"))
   systemProperty("tests.old_cluster_version", "0.0.0")
   onlyIf("tests.bwc.main.version system property exists") { System.getProperty("tests.bwc.main.version") != null }
+  filter {
+    // Mute problematic tests
+    excludeTestsMatching("org.elasticsearch.upgrades.FullClusterRestartIT.testSnapshotRestore")
```

Comment:
those class specific filters should be defined on the specific task not within the plugin in general IMO. That makes it easier to detect those filters for other engineers directly in the according build script without the need to dig into the plugin logic.